packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1523,6 +1571,10 @@ class SliverAppBar extends StatefulWidget {
   /// {@macro flutter.material.appbar.backgroundColor}
   ///
   /// This property is used to configure an [AppBar].
+  ///
+  /// If this color is a [MaterialStateColor] it will be resolved against
+  /// [MaterialState.scrolledUnder] when the content of the app's
+  /// primary scrollable overlaps the app bar.
```

Comment:
Oh I think you also need to update the documentation for `AppBar`'s backgroundColor property

Reply:
Good point, will do.